rc/extra/racer.kak

```diff
@@ -111,13 +111,12 @@ define-command racer-go-definition -docstring "Jump to where the rust identifier
           racer_column=$(printf %s\\n "$racer_data" | cut -f4 )
           racer_file=$(printf %s\\n "$racer_data" | cut -f5 )
           printf %s\\n "edit -existing '$racer_file' $racer_line $racer_column"
-          if [[ "$racer_file" == ${RUST_SRC_PATH}* ]]; then
-            printf %s\\n "set-option buffer readonly true" # The definition resides on the standard library, and the new buffer should be readonly
-          else
-            if [[ "$racer_file" == ${HOME}/.cargo/registry/src* ]]; then
-              printf %s\\n "set-option buffer readonly true" # The definition resides on an external crate, and the new buffer should be readonly
-            fi
-          fi
+          case ${racer_file} in
+            "${RUST_SRC_PATH}"*)
+              printf %s\\n "set-option buffer readonly true";; # The definition resides on the standard library, and the new buffer should be readonly
+            "${HOME}/.cargo/registry/src/"*)
```

Comment:
Cargo also supports `$CARGO_HOME` to override the `~/.cargo` directory, so you should check for that too.

Also, a `case` pattern can be composed of multiple patterns separated by `|`, so you could say:

    "${RUST_SRC_PATH}"* | "${CARGO_HOME:-$HOME/.cargo}/registry/src/*)
        printf %s\\n yada yada
        ;;